superset/security.py

```diff
@@ -410,8 +412,12 @@ def set_perm(self, mapper, connection, target):  # noqa
                 .values(perm=target.get_perm()),
             )
 
-        # add to view menu if not already exists
         permission_name = 'datasource_access'
+        from superset.models.core import Database
+        if mapper.class_ == Database:
```

Comment:
FYI , `superset init` would create the proper `database_access` (one for each database), but the code here (executed on update/insert) would create/update a `datasource_access` entry for the database